plugins/inputs/docker/docker.go

```diff
@@ -435,6 +435,22 @@ func (d *Docker) gatherContainer(
 			}
 		}
 	}
+	if info.State != nil {
+		statefields := map[string]interface{}{
+			"status":     info.State.Status,
+			"running":    info.State.Running,
+			"paused":     info.State.Paused,
+			"restarting": info.State.Restarting,
+			"oomkilled":  info.State.OOMKilled,
+			"dead":       info.State.Dead,
+			"pid":        info.State.Pid,
+			"exitcode":   info.State.ExitCode,
+			"error":      info.State.Error,
+			"startedat":  info.State.StartedAt,
+			"finishedat": info.State.FinishedAt,
```

Comment:
I'm unsure if error, startedat, finishedat are interesting enough to include, I feel like many people will want to exclude them.

Reply:
I feel startedat,finishedat is important since it gives useful information on the uptime of the container or even how long ago the container died(based on "status") . If you still think its not needed , we can take it off!

Reply:
Regarding error . I will remove it..

Reply:
Okay, lets just rename them `started_at` and `finished_at` and convert them to unixnano format (unix epoch timestamp in nanosecond precision).  This is the format we have been using recently for encoding timestamps into a field.

Reply:
I've renamed this to started_at and finished_at . 
I have a query regarding converting the time from RFC3339 format (Example : 2018-06-14T05:48:53.266176036Z ) to Unix Nano format . Golang has the option to convert using the following code : 
``
t1, err := time.Parse(time.RFC3339,"0001-01-01T00:00:00Z")
``
and then use 
``
t1.UnixNano()
``
Here's some sample output of the timestamps : 

> docker_container_status,container_image=ubuntu,container_name=vibrant_benz,container_status=running,container_version=unknown,deviceid=mydeviceid,engine_host=raspberrypi,server_version=18.05.0-ce started_at="2018-06-14T05:48:53.266176036Z",finished_at="0001-01-01T00:00:00Z",oomkilled=false,pid=5278i,exitcode=0i 1528956922000000000
> 
In the above case the docker is still running so it has a valid start time , although its endtime is invalid which is "0001-01-01T00:00:00Z" . If i use the above go code to convert this to unix nano , I get "-6795364578871345152" . As you can see it returns a long negative invalid value . Would this be okay? How else would you like me to handle this ?If this is fine then I will go ahead and commit this

Reply:
I think you should be able to use time.IsZero(), and if true skip adding the field.

Reply:
Thanks ! I've done that.